### PR TITLE
Handle country serialisation with inbuilt functions

### PR DESCRIPTION
**Pull request by ramyaragupathy**
Per discussion updated the country serialisation approach

### CODE REVIEW

File: backend/models/postgis/project.py [hunk 1]
```diff
@@ -271,8 +271,8 @@ def set_country_info(self):
         url = "https://nominatim.openstreetmap.org/reverse?format=jsonv2&lat={0}&lon={1}".format(
             lat, lng
         )
-        country_info = requests.get(url)
-        country_info_json = country_info.content.decode("utf8").replace("'", '"')
+        country_info = requests.get(url).json()  # returns a dict
+        country_info_json = json.dumps(country_info)  # dict -> json
         # Load the JSON to a Python list & dump it back out as formatted JSON
         data = json.loads(country_info_json)
```
Comment (willemarcel) [inline backend/models/postgis/project.py:277]: you can remove the `json.dumps` and the `json.loads`. `country_info["address"].get("country")` works.

Comment (JorgeMartinezG) [reply backend/models/postgis/project.py:277]: Agree. Basically the change should be
```data = requests.get(url).json()```

and remove line 277

**Review by willemarcel — APPROVED** [review]
@ramyaragupathy @JorgeMartinezG I added a commit with a better error handling.

It won't fail if nominatim is offline or if the request doesn't return country information (geocoding failure).

**Comment by willemarcel** [comment]
I made another PR with
- an improvement on the query
- if the project has not country information, fetch nominatim again when the project is updated.

**Comment by ramyaragupathy** [comment]
Tested PR with a polygon that could not be geocoded to a country location and this works good.

**Review by willemarcel — APPROVED** [review]

**Pull request merged**